usecases/schema/class.go

```diff
@@ -75,29 +78,115 @@ func (h *Handler) GetCachedClass(ctxWithClassCache context.Context,
 	}
 
 	return classcache.ClassesFromContext(ctxWithClassCache, func(names ...string) (map[string]versioned.Class, error) {
-		vclasses, err := h.schemaManager.QueryReadOnlyClasses(names...)
-		if err != nil {
-			return nil, err
+		return h.classGetter(h, names)
+	}, names...)
+}
+
+type classGetter func(h *Handler, names []string) (map[string]versioned.Class, error)
```

Comment:
Should I make this an interface instead? If so, I'll have to think about what (if any) fields the struct which implements the interface should have...